requirements-dev.txt

```diff
@@ -7,4 +7,4 @@ sphinx-autobuild
 sphinx-click
 sphinxcontrib-fulltoc
 sphinxcontrib-napoleon
-yapf
+yapf==0.32.0
```

Comment:
Pinning this helps keep CI builds reproducible and free of surprise breakage from upstream.

Reply:
I go back and forth on this, but lately I've been **not** pinning dev dependencies in order to make sure we pick up the "latest and greatest" -- I guess my thought is that tooling/linting surprise breakages won't break downstream users and are generally easy fixes. But it's not a strongly held opinion :-).